main.go

```diff
@@ -14,39 +14,56 @@ import (
 	"github.com/orbs-network/orbs-network-go/config"
 	"github.com/orbs-network/orbs-network-go/instrumentation"
 	"github.com/orbs-network/orbs-network-go/synchronization/supervised"
+	"github.com/orbs-network/scribe/log"
+	"github.com/pkg/errors"
 	"os"
 )
 
 func main() {
-	httpAddress := flag.String("listen", ":8080", "ip address and port for http server")
-	silentLog := flag.Bool("silent", false, "disable output to stdout")
-	pathToLog := flag.String("log", "", "path/to/node.log")
-	version := flag.Bool("version", false, "returns information about version")
+	logger := instrumentation.GetBootstrapCrashLogger()
+	var node *bootstrap.Node
+	func() { // context of bootstrap crash logging
+		defer func() {
+			if r := recover(); r != nil {
+				logger.Error("unexpected error during bootstrap", log.Error(errors.Errorf("unknown error: %v", r)))
+				os.Exit(1)
+			}
+		}()
+		httpAddress := flag.String("listen", ":8080", "ip address and port for http server")
+		silentLog := flag.Bool("silent", false, "disable output to stdout")
+		pathToLog := flag.String("log", "", "path/to/node.log")
+		version := flag.Bool("version", false, "returns information about version")
 
-	var configFiles config.ArrayFlags
-	flag.Var(&configFiles, "config", "path/to/config.json")
+		var configFiles config.ArrayFlags
+		flag.Var(&configFiles, "config", "path/to/config.json")
 
-	flag.Parse()
+		flag.Parse()
 
-	if *version {
-		fmt.Println(config.GetVersion())
-		return
-	}
+		if *version {
+			fmt.Println(config.GetVersion())
+			return
+		}
 
-	cfg, err := config.GetNodeConfigFromFiles(configFiles, *httpAddress)
-	if err != nil {
-		fmt.Printf("%s \n", err)
-		os.Exit(1)
-	}
+		cfg, err := config.GetNodeConfigFromFiles(configFiles, *httpAddress)
+		if err != nil {
+			logger.Error("error reading configuration", log.Error(err))
+			os.Exit(1)
+		}
 
-	logger := instrumentation.GetLogger(*pathToLog, *silentLog, cfg)
+		logger = instrumentation.GetLogger(*pathToLog, *silentLog, cfg)
 
-	node := bootstrap.NewNode(
-		cfg,
-		logger,
-	)
-
-	supervised.NewShutdownListener(logger, node).ListenToOSShutdownSignal()
+		node = bootstrap.NewNode(
+			cfg,
+			logger,
+		)
 
+		supervised.NewShutdownListener(logger, node).ListenToOSShutdownSignal()
+	}()
+	defer func() {
+		if r := recover(); r != nil {
+			logger.Error("unexpected error during runtime", log.Error(errors.Errorf("unknown error: %v", r)))
```

Comment:
as this goroutine is blocked on `WaitUntilShutdown`, I don't see how any panic can be thrown from it. It's not as if this `defer func()` is a "catch all exception handler".

Reply:
Agree.

errors happen: out of memory, seg faults, sneaky edge case bugs in 3rd party libraries etc. If they happen in the main routine, I'd like to log them to the right output (configured in the logger) before the server dies.